net.c

```diff
@@ -328,6 +328,22 @@ int redisContextSetTimeout(redisContext *c, const struct timeval tv) {
     return REDIS_OK;
 }
 
+static int _redisContextUpdateTimeout(redisContext *c, const struct timeval *timeout) {
+    /* Same timeval struct, short circuit */
+    if (c->timeout == timeout)
```

Comment:
@michael-grunder You probably meant to compare the actual value and not the pointer here.